spec/models/user_spec.rb

```diff
@@ -114,4 +114,45 @@
       expect(user2.user_id).to be > user.user_id
     end
   end
+
+  describe 'notifying the user' do
+    let(:user) { FactoryBot.create :user }
+
+    def notify_user
+      user.notify 'my notification', 'some message', 'https://www.google.com/'
+    end
+
+    it 'creates a notification object' do
+      expect { notify_user }.to change(Notification, :count).by(1)
+    end
+
+    it 'notifies slack' do
+      allow(user).to receive(:notify_slack)
+      notify_user
+      expect(user).to have_received(:notify_slack)
+    end
+
+    it 'does not create multiple duplicate notifications' do
+      3.times { notify_user }
+      expect(user.notifications.size).to be == 1
```

Comment:
`.to eql 1` ?

Reply:
According to: [Equality matchers](https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/equality-matchers)
You should use expect(a).to be == b if you want to check if a == b is true.
With numbers I would always use == and not eql?, so from my viewpoint this is the correct matcher

Reply:
`a.eql?(b)`: a and b have the same value
`a == b`: a and b have the same value _with type conversions_

Both are probably okay here. However, I would suggest:
`expect(a).to eq(b) # passes if a == b` which reads a little better, imho.

Reply:
I think `expect(a).to eq(b)` is too similar to `expect(a).to eql(b)`. The one letter difference can be easily overlooked. So I do not want to encourage usage of `.to eq` anywhere.
Especially as there usually is no a.eq?(b) method, so I would not know what `expect(a).to eq(b)` does. 

`expect(a).to be == b` may be a bit of an unusual syntax, but it clearly shows that == is used for the equality check. Which is the appropriate operation for equality checks on numbers.

Reply:
Good points. `eq` is the standard throughout the current code base.